// @atlaskit/tag/dist/esm/tag/internal/shared/styles.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import { buttonWidthUnitless, defaultBorderRadius, defaultMargin, defaultRoundedBorderRadius, defaultTextPadding, maxTextWidth, maxTextWidthUnitless, tagHeight, textFontSize, textPaddingLeft, textPaddingRight } from '../../../constants';
export var roundedBorderStyles = {
  borderRadius: "".concat(defaultRoundedBorderRadius)
};
export var beforeElementStyles = {
  position: 'absolute',
  left: 0,
  borderRadius: "".concat(defaultBorderRadius),
  alignItems: 'center',
  display: 'flex',
  justifyContent: 'center'
};
export var chromeStyles = function chromeStyles(_ref) {
  var backgroundColor = _ref.backgroundColor,
      backgroundColorHover = _ref.backgroundColorHover,
      textColor = _ref.textColor,
      textColorHover = _ref.textColorHover;
  return {
    backgroundColor: backgroundColor,
    color: textColor,
    '&:hover': {
      backgroundColor: backgroundColorHover,
      color: textColorHover
    },
    '& button:hover': {
      backgroundColor: backgroundColor,
      color: textColor
    },
    '&>span:hover': {
      backgroundColor: backgroundColor,
      color: textColor
    },
    borderRadius: "".concat(defaultBorderRadius),
    cursor: 'default',
    display: 'inline-flex',
    position: 'relative',
    height: "".concat(tagHeight),
    lineHeight: 1,
    margin: "".concat(defaultMargin),
    padding: 0,
    overflow: 'hidden'
  };
};
export var chromeLinkStyles = function chromeLinkStyles(_ref2) {
  var hoverBackgroundColor = _ref2.hoverBackgroundColor,
      hoverTextColor = _ref2.hoverTextColor,
      hoverBackgroundColorRemoval = _ref2.hoverBackgroundColorRemoval,
      hoverTextColorRemoval = _ref2.hoverTextColorRemoval,
      focusRingColor = _ref2.focusRingColor;
  return {
    '&>a:hover': {
      backgroundColor: hoverBackgroundColor,
      color: hoverTextColor
    },
    '&[data-removable="true"]:hover': {
      backgroundColor: hoverBackgroundColorRemoval,
      color: hoverTextColorRemoval
    },
    '&:focus-within': {
      boxShadow: "0 0 0 2px ".concat(focusRingColor),
      outline: 'none'
    },
    '&[data-removing="true"]:focus-within': {
      boxShadow: "0 0 0 2px transparent",
      outline: 'none'
    }
  };
};
export var hasAfterStyles = {
  paddingRight: "".concat(textPaddingRight),
  maxWidth: "".concat(maxTextWidthUnitless - buttonWidthUnitless, "px")
};
export var hasBeforeStyles = {
  paddingLeft: "".concat(textPaddingLeft)
};
export var textStyles = {
  fontSize: textFontSize,
  fontWeight: 'normal',
  lineHeight: 1,
  paddingLeft: defaultTextPadding,
  paddingRight: defaultTextPadding,
  paddingTop: '2px',
  paddingBottom: '2px',
  borderRadius: defaultBorderRadius,
  maxWidth: maxTextWidth,
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap'
};
export var linkStyles = function linkStyles(linkHoverColor) {
  return _objectSpread(_objectSpread({}, textStyles), {}, {
    ':not([data-color="standard"])': {
      color: 'inherit',
      textDecoration: 'underline'
    },
    textDecoration: 'none',
    '&:hover': {
      color: linkHoverColor
    },
    '&:focus': {
      outline: 'none'
    }
  });
};
