// @atlaskit/tag/dist/cjs/tag/internal/removable/styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.removeButtonStyles = void 0;

var _colors = require("@atlaskit/theme/colors");

var _constants = require("../../../constants");

// NOTE:
// "-moz-focus-inner" removes some inbuilt padding that Firefox adds (taken from reduced-ui-pack)
// the focus ring is red unless combined with hover, then uses default blue
var removeButtonStyles = function removeButtonStyles(_ref) {
  var backgroundColor = _ref.backgroundColor,
      backgroundColorHover = _ref.backgroundColorHover,
      focusBoxShadowColor = _ref.focusBoxShadowColor,
      hoverBoxShadowColor = _ref.hoverBoxShadowColor;
  return {
    position: 'absolute',
    right: 0,
    backgroundColor: backgroundColor,
    alignItems: 'center',
    alignSelf: 'center',
    appearance: 'none',
    border: 'none',
    borderRadius: _constants.defaultBorderRadius,
    color: _colors.N500,
    display: 'flex',
    justifyContent: 'center',
    height: '16px',
    margin: 0,
    padding: 0,
    cursor: 'pointer',
    '&::-moz-focus-inner': {
      border: 0,
      margin: 0,
      padding: 0
    },
    '&:focus': {
      boxShadow: "0 0 0 2px ".concat(focusBoxShadowColor, " inset"),
      outline: 'none'
    },
    '&:hover': {
      backgroundColor: "".concat(backgroundColorHover, " !important"),
      color: _colors.R500
    }
  };
};

exports.removeButtonStyles = removeButtonStyles;
